packages/@aws-cdk/aws-certificatemanager/lambda-packages/dns_validated_certificate_handler/lib/index.js

```diff
@@ -99,42 +99,43 @@ const requestCertificate = async function(requestId, domainName, subjectAlternat
 
   console.log('Waiting for ACM to provide DNS records for validation...');
 
-  let record;
-  for (let attempt = 0; attempt < maxAttempts && !record; attempt++) {
+  let options = [];
+  let ready = false;
+  for (let attempt = 0; attempt < maxAttempts && !ready; attempt++) {
     const { Certificate } = await acm.describeCertificate({
       CertificateArn: reqCertResponse.CertificateArn
     }).promise();
-    const options = Certificate.DomainValidationOptions || [];
+    options = Certificate.DomainValidationOptions || [];
 
-    if (options.length > 0 && options[0].ResourceRecord) {
-      record = options[0].ResourceRecord;
-    } else {
+    ready = options.length && options.every(opt => opt && !!opt.ResourceRecord);
+    if (!ready) {
       // Exponential backoff with jitter based on 200ms base
       // component of backoff fixed to ensure minimum total wait time on
       // slow targets.
       const base = Math.pow(2, attempt);
       await sleep(random() * base * 50 + base * 150);
     }
   }
-  if (!record) {
-    throw new Error(`Response from describeCertificate did not contain DomainValidationOptions after ${maxAttempts} attempts.`)
+  if (!ready) {
+    throw new Error(`Response from describeCertificate did not contain all DomainValidationOptions after ${maxAttempts} attempts.`);
   }
 
-  console.log(`Upserting DNS record into zone ${hostedZoneId}: ${record.Name} ${record.Type} ${record.Value}`);
-
   const changeBatch = await route53.changeResourceRecordSets({
     ChangeBatch: {
-      Changes: [{
-        Action: 'UPSERT',
-        ResourceRecordSet: {
-          Name: record.Name,
-          Type: record.Type,
-          TTL: 60,
-          ResourceRecords: [{
-            Value: record.Value
-          }]
-        }
-      }]
+      Changes: options.map(({ ResourceRecord: record }) => {
+        console.log(`Upserting DNS record into zone ${hostedZoneId}: ${record.Name} ${record.Type} ${record.Value}`);
+        return {
+          Action: 'UPSERT',
+          ResourceRecordSet: {
+            Name: record.Name,
+            Type: record.Type,
+            TTL: 60,
+            ResourceRecords: [{
+              Value: record.Value
+            }]
+          }
+        };
+      })
```

Comment:
Can you explain why this change in logic is needed in this PR?

Reply:
As @saltman424 mentioned below and as it is pointed out in #4659, current logic only creates a single Route53 validation record and misses creating all other records for subjectAlternativeNames.